### PR TITLE
Internal: Create a single content template for the Nested Tabs [ED-14924]

### PR DESCRIPTION
**Pull request by hein-obox**
## PR Checklist

- [ ] The commit message follows our guidelines:  https://github.com/elementor/elementor/blob/master/.github/CONTRIBUTING.md

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## Summary

This PR can be summarized in the following changelog entry:

*

## Description
An explanation of what is done in this PR

*

## Test instructions
This PR can be tested by following these steps:

*

## Quality assurance

- [ ] I have tested this code to the best of my abilities
- [ ] I have added unittests to verify the code works as intended
- [ ] Docs have been added / updated (for bug fixes / features)

Fixes #

### CODE REVIEW

File: modules/nested-tabs/widgets/nested-tabs.php [hunk 1]
```diff
@@ -1235,120 +1247,88 @@ protected function get_initial_config(): array {
...
+			'class': [ 'e-n-tab-title',hoverAnimationClass ],
 			'data-tab-index': tabCount,
 			'role': 'tab',
 			'aria-selected': 1 === tabCount ? 'true' : 'false',
 			'tabindex': 1 === tabCount ? '0' : '-1',
-			'aria-controls': 'e-n-tab-content-' + elementUid,
+			'aria-controls': 'e-n-tab-content-' + tabUid,
 			'style': '--n-tabs-title-order: ' + tabCount + ';',
-		};
+		}, null, true );
```
Comment (hein-obox) [inline modules/nested-tabs/widgets/nested-tabs.php:1306]: @davseve @louiswol94 
I would like to create a single template inside the editor.

I am not sure how to handle these render attributes. 

From my testing it seems safe to to apply `null, true` to both the content template and the single repeater content template, but I not 100% about this yet.

Wdyt?

File: modules/nested-tabs/widgets/nested-tabs.php [hunk 2]
```diff
@@ -1235,120 +1247,85 @@ protected function get_initial_config(): array {
...
+			tabId = item.element_id
+				? item.element_id
+				: 'e-n-tab-title-' + elementUid + ( tabIndex + 1 ),
+			tabUid = elementUid + tabCount,
+			tabIcon = elementor.helpers.renderIcon( view, item.tab_icon, { 'aria-hidden': true }, 'i' , 'object' ),
+			activeTabIcon = item.tab_icon_active.value
+				? elementor.helpers.renderIcon( view, item.tab_icon_active, { 'aria-hidden': true }, 'i' , 'object' )
+				: tabIcon;
+
+		view.addRenderAttribute( 'e-n-tab-title', {
```
Comment (hein-obox) [inline modules/nested-tabs/widgets/nested-tabs.php:1294]: Now that we overwrite the attribute, we don't need a unique attribute key anymore.

File: modules/nested-tabs/widgets/nested-tabs.php [hunk 3]
```diff
@@ -1235,120 +1247,85 @@ protected function get_initial_config(): array {
...
+			tabId = item.element_id
+				? item.element_id
+				: 'e-n-tab-title-' + elementUid + ( tabIndex + 1 ),
+			tabUid = elementUid + tabCount,
+			tabIcon = elementor.helpers.renderIcon( view, item.tab_icon, { 'aria-hidden': true }, 'i' , 'object' ),
+			activeTabIcon = item.tab_icon_active.value
+				? elementor.helpers.renderIcon( view, item.tab_icon_active, { 'aria-hidden': true }, 'i' , 'object' )
+				: tabIcon;
+
+		view.addRenderAttribute( 'tab-title', {
```
Comment (hein-obox) [inline modules/nested-tabs/widgets/nested-tabs.php:1294]: Now that we overwrite the attribute, we don't need a unique attribute key anymore. We can use normal strings for the attribute keys, like `tab-title`.

File: modules/nested-tabs/widgets/nested-tabs.php [hunk 4]
```diff
@@ -1235,120 +1247,85 @@ protected function get_initial_config(): array {
...
-			tabIcon = elementor.helpers.renderIcon( view, data.tab_icon, { 'aria-hidden': true }, 'i' , 'object' );
-
-		let tabActiveIcon = tabIcon,
-			tabId = 'e-n-tab-title-' + elementUid;
+		const tabIndex = view.collection.length,
+			elementUid = view.getIDInt().toString().substr( 0, 3 ),
+			item = data,
+			hoverAnimationClass = view?.container?.settings?.attributes?.hover_animation
+				? `elementor-animation-${ view.container.settings.attributes.hover_animation }`
+				: '';
```
Comment (davseve) [inline modules/nested-tabs/widgets/nested-tabs.php:1255]: MaybeL 
hoverAnimation = view?.container?.settings?.attributes?.hover_animation;
hoverAnimationClass = hoverAnimation ? `elementor-animation-${hoverAnimation}` : '';

Comment (davseve) [inline modules/nested-tabs/widgets/nested-tabs.php:1255]: Maybe:
```suggestion
hoverAnimation = view?.container?.settings?.attributes?.hover_animation;
hoverAnimationClass = hoverAnimation ? `elementor-animation-${hoverAnimation}` : '';
```

File: modules/nested-tabs/widgets/nested-tabs.php [hunk 5]
```diff
@@ -1235,120 +1247,85 @@ protected function get_initial_config(): array {
...
+		?>
+		<# const elementUid = view.getIDInt().toString().substr( 0, 3 ); #>
+		<div class="e-n-tabs" data-widget-number="{{ elementUid }}" aria-label="<?php echo esc_html__( 'Tabs. Open items with Enter or Space, close with Escape and navigate using the Arrow keys.', 'elementor' ); ?>">
+			<# if ( settings['tabs'] ) { #>
+			<div class="e-n-tabs-heading" role="tablist">
+				<# _.each( settings['tabs'], function( item, index ) {
+					const tabIndex = index,
+					hoverAnimationClass = settings['hover_animation']
+						? `elementor-animation-${ settings['hover_animation'] }`
+						: '';
```
Comment (davseve) [inline modules/nested-tabs/widgets/nested-tabs.php:1270]: Same as above.
Just a suggestion.

File: tests/playwright/sanity/modules/nested-tabs/nested-tabs.test.ts-snapshots/tabs-horizontal-scroll-initial-frontend-linux.png
Comment (davseve) [inline tests/playwright/sanity/modules/nested-tabs/nested-tabs.test.ts-snapshots/tabs-horizontal-scroll-initial-frontend-linux.png:1]: Was this image deleted?

File: modules/nested-tabs/widgets/nested-tabs.php [hunk 6]
```diff
@@ -1106,15 +1123,10 @@ protected function render_tab_titles_html( $item_settings ): string {
...
-					<?php echo $icon_html; // phpcs:ignore WordPress.Security.EscapeOutput.OutputNotEscaped ?>
-					<?php echo $icon_active_html; // phpcs:ignore WordPress.Security.EscapeOutput.OutputNotEscaped ?>
-				</span>
-				<span <?php echo $text_class; // phpcs:ignore WordPress.Security.EscapeOutput.OutputNotEscaped ?>>
-					<?php echo wp_kses_post( $title ); ?>
-				</span>
-			</button>
+		<span <?php $this->print_render_attribute_string( 'tab-icon' ); ?>>
+			<?php echo $icon_html; // phpcs:ignore WordPress.Security.EscapeOutput.OutputNotEscaped ?>
+			<?php echo $icon_active_html; // phpcs:ignore WordPress.Security.EscapeOutput.OutputNotEscaped ?>
```
Comment (davseve) [inline modules/nested-tabs/widgets/nested-tabs.php:1128]: Should these be escaped?

File: modules/nested-tabs/widgets/nested-tabs.php (hunk 4 above)
Comment (hein-obox) [reply modules/nested-tabs/widgets/nested-tabs.php]: @davseve Updated.

File: modules/nested-tabs/widgets/nested-tabs.php (hunk 5 above)
Comment (hein-obox) [reply modules/nested-tabs/widgets/nested-tabs.php]: @davseve Updated.

File: tests/playwright/sanity/modules/nested-tabs/nested-tabs.test.ts-snapshots/tabs-horizontal-scroll-initial-frontend-linux.png
Comment (hein-obox) [reply tests/playwright/sanity/modules/nested-tabs/nested-tabs.test.ts-snapshots/tabs-horizontal-scroll-initial-frontend-linux.png:1]: We need to update this screenshot.
It was updated in the previous PR.

File: modules/nested-tabs/widgets/nested-tabs.php (hunk 6 above)
Comment (hein-obox) [reply modules/nested-tabs/widgets/nested-tabs.php]: We have created a separate task for that: https://elementor.atlassian.net/browse/ED-14938

File: modules/nested-tabs/widgets/nested-tabs.php (hunk 3 above)
Comment (davseve) [reply modules/nested-tabs/widgets/nested-tabs.php:1296]: Don't we need a unique attribute on the widget initial load?

**Review by davseve — CHANGES_REQUESTED** [review]

File: modules/nested-tabs/widgets/nested-tabs.php [hunk 7]
```diff
@@ -1247,129 +1247,90 @@ protected function get_initial_config(): array {
...
 		<#
-		const tabCount = view.collection.length + 1,
-			elementUid = view.getIDInt().toString().substring( 0, 3 ) + tabCount,
-			tabIcon = elementor.helpers.renderIcon( view, data.tab_icon, { 'aria-hidden': true }, 'i' , 'object' );
-
-		let tabActiveIcon = tabIcon,
-			tabId = 'e-n-tab-title-' + elementUid;
+		const tabIndex = view.collection.length,
+			elementUid = view.getIDInt().toString().substr( 0, 3 ),
+			item = data,
```
[the remark this answers is not in the document]
Comment (hein-obox) [reply modules/nested-tabs/widgets/nested-tabs.php:1252]: The template breaks if we skip this step.
Moustache expects the item variable name. We can't pass it as another variable name.

File: modules/nested-tabs/widgets/nested-tabs.php (hunk 1 above)
Comment (hein-obox) [reply modules/nested-tabs/widgets/nested-tabs.php:1305]: @davseve We have set the `overwrite` value to `true`.

File: modules/nested-tabs/assets/js/frontend/handlers/nested-tabs.js [hunk 8]
```diff
@@ -409,6 +417,15 @@ export default class NestedTabs extends Base {
 			this.updateListeners( view );
 			elementor.$preview[ 0 ].contentWindow.dispatchEvent( new CustomEvent( 'elementor/elements/link-data-bindings' ) );
 		}
+
+		this.resetActiveTab();
+	}
+
+	resetActiveTab() {
+		const activeTabIndex = this.getActiveTabIndex() || 0;
```
Comment (hein-obox) [inline modules/nested-tabs/assets/js/frontend/handlers/nested-tabs.js:425]: In some cases, the activate tab wasn't working correctly after using a repeater function.

**Review by davseve — DISMISSED** [review]

File: modules/nested-tabs/widgets/nested-tabs.php [hunk 9]
```diff
@@ -1247,129 +1247,91 @@ protected function get_initial_config(): array {
...
+		const tabCount = tabIndex + 1,
+			tabId = item.element_id
+				? item.element_id
+				: 'e-n-tab-title-' + elementUid + ( tabIndex + 1 ),
+			tabUid = elementUid + tabCount,
+			tabIcon = elementor.helpers.renderIcon( view, item.tab_icon, { 'aria-hidden': true }, 'i' , 'object' ),
+			activeTabIcon = item.tab_icon_active.value
+				? elementor.helpers.renderIcon( view, item.tab_icon_active, { 'aria-hidden': true }, 'i' , 'object' )
+				: tabIcon,
+			escapedHoverAnimationClass = _.escape( hoverAnimationClass );
```
Comment (hein-obox) [inline modules/nested-tabs/widgets/nested-tabs.php:1295]: @louiswol94 Hover animation class escaped.

**Review by louiswol94 — APPROVED** [review]

**Pull request merged**